Update Docker selfhosting guide (in-progress)

shikhir-arora opened the pull request:
**This is not ready to merge.** 

Just opening PR to track commits. 

- [x] Update Docker guide with `Dev` only
- [ ] More detailed info about config options within the main files 
- [ ] Brief info about new structure (Quarterdeck, single `fredboat.yaml` file) 
- [ ] Organization/cleanup

--

Input is welome!

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/FredBoat/fredboat.com?pullRequest=28) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/FredBoat/fredboat.com?pullRequest=28) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/FredBoat/fredboat.com?pullRequest=28) it.</sub>

shikhir-arora commented:
/cc @Frederikam @napstr and anyone else - feel free to input any other suggestions. Should be ready to merge; I've been busy and just got around to modifying the rest of the guide.

I removed information about the `stable` install because that likely isn't going to work at all in the future (using the old `credentials`/`config` style) and it is a bit less confusing if there's just one style of install, IMO.

I also emphasized that it is strongly suggested to use Docker. I've upgraded the links to QuarterDeck, FredBoat/Dev, and the requisite files. Also updated the images to match the changes.

napstr commented on public/markdown/selfhosting.md:
```diff
@@ -25,12 +25,12 @@ FredBoat requires a PostgreSQL database to run. The easiest way to achieve that
 - Docker works on almost any architecture and platform out there in the same way - be it Windows, Mac OS, tons of Linux distros or even your spare Raspberry Pi.
 - You don't have to install Java or any build tools like git or gradle, and copy paste obscure commands to have them spit out a jar, because our [CI server](https://ci.fredboat.com) builds the jar for you and publishes an official FredBoat docker image to the [official docker hub](https://hub.docker.com/r/fredboat/fredboat/).
-- You don't have to set up, configure and secure a [PostgreSQL database](https://www.postgresql.org/) which is required by FredBoat, because we built a docker image for that and docker-compose will set it up and start it together with FredBoat and keep it running.
+- You don't have to set up, configure and secure a [QuarterDeck database](https://github.com/FredBoat/Backend) which is required by FredBoat, because we built a docker image for that and docker-compose will set it up and start it together with FredBoat and keep it running.
```
If they aren't kept up-to-date, they should probably be removed. I would recommend it, but it's not my call to make.
What I can say is that I've been trying to have all our projects produce their jars with `./gradlew build`, so if you're feeling generous, this could be a thing worth mentioning. It is _probably_ better to just refer to the READMEs and build scripts though.

shikhir-arora commented:
Do you think putting it on another page -- called "unsupported" or something -- should be the way to go?

OtterBoops commented:
I am going to chime in here and say that "Unsupported" pages are probably not a good thing as links/builds/containers could be deprecated, which would make the situation look even more unmaintained.

shikhir-arora commented:
Yeah I didn't mean it that way -- Let me take the change requests in account and I'll push some stuff.

shikhir-arora commented:
*Way* more commits than any normal human should have needed, but let me know how it looks 👍

Frederikam commented on public/markdown/selfhosting.md:
```diff
@@ -97,96 +103,52 @@ Congratulations! Your FredBoat should be running now. If you haven't invited you
 
-If you want to do any changes to FredBoat's code, this is the place and time to do them. Remember, after any changes you need to run the build again to compile a fresh jar file containing your changes.
 
+- From inside the `FredBoat` folder downloaded from GitHub, you can now compile FredBoat using Gradle 4.7+ (it is always a good idea to ensure you have the proper Gradle version(s)):
```
This version number will become outdated. We include Gradle wrappers for Windows and *NIX. I personally let IDEA use the wrappers, and I don't have gradle globally installed.

shikhir-arora replied:
Fixed.

Frederikam replied:
:+1:

shikhir-arora commented:
Let me know if anything else is needed, otherwise I think @napstr stated he's good with it. My goal was to make the case for Docker much stronger, and then include the binary section at the end -- but not a full tutorial and startup guide on that, just so users could be aware that they can still compile the jar's with Gradle - per Napster's comments. So I think it's pretty clear now: there's not many ways to install so it's one path and only dev build so less confusion between multiple ones. Many selfhosters often tend to choose the "lower level" methods (like source install) sometimes for the fun of it -- I was one of those people back then -- but in this case there's a strong reason not to run it completely from source, and hopefully the benefits are outlined succinctly!

shikhir-arora commented:
[How's that?](https://5af064dc0733d575903c7207--fredboatcom.netlify.com/docs/selfhosting)?

Frederikam commented on public/markdown/selfhosting.md:
````diff
@@ -196,42 +155,34 @@ or on Windows:
-```
-
-Dev:
 ```sh
````
The trailing linebreaks in this code block look weird

shikhir-arora replied:
Removed those ✔️

Frederikam commented on public/markdown/selfhosting.md:
```diff
@@ -2,7 +2,7 @@
 * * *
 
-**Warning**: This guide is a community effort, and has not been updated to the latest version of FredBoat by the selfhosting community. You will not be able to run the latest code with the current state of this document. You can help update / improve this guide by submitting your pull requests [here]( https://github.com/FredBoat/fredboat.com/blob/master/public/markdown/selfhosting.md).
+**Disclaimer**: This guide is a highly driven community effort, and as such is always subject to change -- sometimes breaking. Make sure to check the community-based #selfhosters channel in the [FredBoat Hangout](https://discord.gg/cgPFW4q) server and, as always, pull requests are welcome [here]( https://github.com/FredBoat/fredboat.com/blob/master/public/markdown/selfhosting.md).
```
"driven community" should be "community-driven".

Selfhosters *may* want to check the selfhosting channel, but not always.

shikhir-arora replied:
✅ I also changed the #selfhosting channel reference from the guide in another section to the proper #selfhosters 🙂

Frederikam commented:
Thanks a lot!

@napstr @OtterBoops do you have any last remarks before merging?

napstr commented:
Any remarks to what? I'm happy to answer concrete, technical questions to running different parts of FredBoat.

Frederikam commented:
>Any remarks to what?

As in do you have anything else you'd like changed?

napstr commented:
No. As stated repeatedly in the past, I have no interest in how the selfhosting documentation looks. Please do not consult me about it.
